### PR TITLE
UNI-1060 Network select missing member details

### PR DESCRIPTION
**Pull request by 0xbarbs**
Various improvements to the network select component:

- If account not member, button now says “Begin Membership Process”
- Each network shows member status and available DAI
- Added becoming a member docs links

### CODE REVIEW

File: src/providers/MemberData.jsx [hunk 1]
```diff
@@ -207,6 +207,44 @@ export function useMemberData(address, chainId) {
   };
 }
 
+export function useMemberSummary(address, chainId) {
```
Comment (GeraldHost) [inline src/providers/MemberData.jsx]: Lets separate this out into it's own file in `hooks`

File: src/components/connect/NetworkSelect.jsx [hunk 2]
```diff
@@ -67,7 +70,7 @@ export default function NetworkSelect() {
           <Box fluid align="center" direction="vertical" mb="6px">
             {networks.map((props) => {
               const { label, value, avatar, description, chainId } = props;
-
+              const { data } = useMemberSummary(address, chainId);
```
Comment (GeraldHost) [inline src/components/connect/NetworkSelect.jsx]: Calling hooks inside loops is a bit of an anti pattern, if the length of `networks` changes over re-renders then the app will break. Obvs networks is constant here but better to not use hooks like this. Instead create a new component `NetworkSelectOption` and encapsulate the logic there. This should be a new file in `components` and any CSS that is in `NetworkSelect.scss` that is relevant to this component to `NetworkSelectOption.scss` and rename the classes so they follow the convention.

File: src/components/connect/NetworkSelect.jsx [hunk 3]
```diff
@@ -65,60 +77,15 @@ export default function NetworkSelect() {
...
-                  </Box>
-                </Card>
-              );
-            })}
+            {networks.map((network) => (
+              <NetworkSelectOption
+                address={address}
+                network={network}
+                disabled={!isConnected}
+                changeNetwork={handleChangeNetwork}
```
Comment (GeraldHost) [inline src/components/connect/NetworkSelect.jsx]: I would probably change this pattern a little bit. As currently you need to remember in NetworkSelectOption that changeNetwork accepts two network arguments. Instead I think the `NetworkSelectOption` only needs to be concerned with if it is clicked. I would change this to be:

```js
// update this to be a partial application that just takes network as an argument
// as chainId already exists on the network argument so no need to pass it in aswell
const handleChangeNetwork = (network) => () => {
  ...
}

// Then when you render do this:
{networks.map((network) => (
  <NetworkSelectOption
    ...
    onClick={handleChangeNetwork(network)}
    ...
    />
```

Then inside `NetworkSelectOption` you just need to assign `onClick` to the `onClick` prop of the underlying component.

Comment (0xbarbs) [reply src/components/connect/NetworkSelect.jsx]: Good shout, I've made and pushed the suggested change

**Review by GeraldHost — APPROVED** [review]

**Pull request merged**